src/analysis/imports.rs

```diff
@@ -119,21 +120,37 @@ impl Imports {
         self.defaults.clear();
     }
 
+    fn common_checks(&self, name: &str) -> bool {
+        if name == "crate" || name == "ffi" {
+            false
+        } else if let Some(ref defined) = self.defined {
+            !((name.starts_with("crate::") && &name[7..] == defined) || name == defined)
+        } else {
+            true
+        }
+    }
+
     /// Declares that name should be available through its last path component.
     ///
     /// For example, if name is `X::Y::Z` then it will be available as `Z`.
     /// Uses defaults.
     pub fn add(&mut self, name: &str) {
-        if let Some(ref defined) = self.defined {
-            if name == defined {
-                return;
-            }
+        if !self.common_checks(name) {
+            return;
         }
-        if let Some(name) = self.strip_crate_name(name) {
+        if let Some(mut name) = self.strip_crate_name(name) {
+            // Dirty little hack to allow to have correct import for GDKX11.
+            if name == "xlib" {
+                name = if self.crate_name == "gdk_x11" {
+                    Cow::Borrowed("x11::xlib")
+                } else {
+                    Cow::Borrowed("crate::xlib")
```

Comment:
`x11::xlib` will also work from inside `gdkx11`, no need to complicate it (and then you can also remove the `pub(crate) use x11::xlib as xlib` from there)